server/server.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"log"
+	"net"
+	"net/rpc"
+	"sync"
+
+	"github.com/lukedmor/conduit_tiny/common"
+)
+
+type Provider struct {
+	cs *conduitServer
+}
+
+func (p *Provider) Join(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	p.cs.providers[a.Addr] = true
```

Comment:
We shouldn't really be trusting the provider to give its own address — we should probably determine it ourselves. When the provider joins, we should also send them a performance evaluation script for use in resource allocation, if a client has specific needs.